### PR TITLE
build: remove urbit-king haskell releases and haskell.nix dependency

### PR DESCRIPTION
**Pull request by brendanhay**
* Upgrades the nixpkgs pin to 21.11.
* Removes nix code, sources, and attributes related to haskell.nix and  the code under pkg/hs.
* Continuous integration no longer builds (or packages/tarballs/releases) urbit-king (haskell).
* To continue to developing any haskell code under pkg/hs it's left as  a user exercise uto install and configure GHC/Stack.
* Fixes some minor issues like overriding the existing `curlMinimal` attribute in `overlays/native.nix` causing unnecessary rebuilds.

As a consequence running nix-shell should be considerably more performant and requires downloading or building less dependencies. CI also has less work to do in general.

### CODE REVIEW

**Review by joemfb — CHANGES_REQUESTED** [review]
This all looks reasonable. I'm not sure what's going wrong with the docker image ... (/cc @eamsden).

a secp nit:

**Comment by joemfb** [comment]
Also, this works on my machine without nixpkgs version patches, which is pretty nice. This PR supersedes #5165.

**Comment by brendanhay** [comment]
> Also, this works on my machine without nixpkgs version patches, which is pretty nice. This PR supersedes #5165.

Imagine if we updated nixpkgs more than once per year!

**Comment by mrdomino** [comment]
went ahead and added schnorr signatures to the pmnsh secp256k1 build. If desired, can drop that and put it in separately.

LGTM

**Review by mrdomino — APPROVED** [review]

**Comment by mrdomino** [comment]
\> it failed

that's embarrassing. out-of-date commit. will fix.

**Comment by mrdomino** [comment]
i just force-pushed my commits out of your PR, sorry, i'll go and sort them out on my own.

File: nix/sources.json [hunk 1]
```diff
@@ -99,21 +75,39 @@
...
+            "include": "include",
+            "lib": ".libs",
+            "make": "libsecp256k1.la",
+            "prepare": "./autogen.sh && ./configure --disable-shared --enable-module-recovery CFLAGS=-DSECP256K1_API="
+        },
+        "repo": "secp256k1",
+        "rev": "26de4dfeb1f1436dae1fcf17f57bdaa43540f940",
+        "sha256": "03i3nv8d3ci7q9y98q11rrp3rvwdqc0hc0ss0pr6xckybvizsmbb",
+        "type": "tarball",
+        "url": "https://github.com/bitcoin-core/secp256k1/archive/26de4dfeb1f1436dae1fcf17f57bdaa43540f940.tar.gz",
```
[the remark this answers is not in the document]
Comment (brendanhay) [reply nix/sources.json]: Updated PTAL

**Comment by mrdomino** [comment]
Made a PR against your PR: #5483

**Review by joemfb — APPROVED** [review]
LGTM

**Pull request merged**